src/java.base/share/classes/java/lang/classfile/CodeBuilder.java

```diff
@@ -582,6 +593,7 @@ default CodeBuilder newObjectInstruction(ClassEntry type) {
      * Generate an instruction to create a new object
      * @param type the object type
      * @return this builder
+     * @throws IllegalArgumentException if {@code type} represents a primitive type
```

Comment:
What about arrays? (ClassDesc has an `isArray` predicate too)

Reply:
Test for arrays is not performed. The only constraint here is ability to construct constant pool class entry, which is not possible from a primitive type.
The library does only bare minimal consistency checks and proposal for additional constraints should be discussed at the mailing list.